File: app/src/main/java/com/rogger/xcast10/VideoItem.java
package com.rogger.xcast10;

import android.graphics.Bitmap;
import android.net.Uri;

import java.util.Locale;

/**
 * Representa um vídeo local com título, miniatura e URI.
 */
public class VideoItem {

    private final String title;
    private final Uri uri;
    private final Bitmap thumbnail;
    private final long duration;
    private final String path; // ← novo campo

    public VideoItem(String title, Uri uri, Bitmap thumbnail, long duration, String path) {
        this.title = title;
        this.uri = uri;
        this.thumbnail = thumbnail;
        this.duration = duration;
        this.path = path;
    }

    public String getTitle() {
        return title;
    }

    public Uri getUri() {
        return uri;
    }

    public Bitmap getThumbnail() {
        return thumbnail;
    }

    public long getDuration() {
        return duration;
    }

    public String getPath() {
        return path;
    }

    @Override
    public String toString() {
        return title;
    }
    public String formated(){
        return formatDuration(duration);
    }
    private String formatDuration(long durationMs) {
        long seconds = durationMs / 1000;
        long h = seconds / 3600;
        long m = (seconds % 3600) / 60;
        long s = seconds % 60;
        if (h > 0) {
            return String.format(Locale.getDefault(), "%02d:%02d:%02d", h, m, s);
        } else {
            return String.format(Locale.getDefault(), "%02d:%02d", m, s);
        }
    }

}

File: app/src/main/java/com/rogger/xcast10/data/model/VideoItem.kt
package com.rogger.xcast10.data.model

import android.graphics.Bitmap
import android.net.Uri
import java.util.Locale

/*
 * Desenvolvido por Roger de Oliveira
 * Data: 03/07/2026
 * Hora: 12:09
 */
data class VideoItem(
    val title: String,
    val uri: Uri,
    val thumbnail: Bitmap?,
    val duration: Long, // em milissegundos
    val path: String?

){
    /** Duração formatada, ex: "12:03" ou "01:12:03". */
    fun formattedDuration(): String {
        val seconds = duration / 1000
        val h = seconds / 3600
        val m = (seconds % 3600) / 60
        val s = seconds % 60
        return if (h > 0) {
            String.format(Locale.getDefault(), "%02d:%02d:%02d", h, m, s)
        } else {
            String.format(Locale.getDefault(), "%02d:%02d", m, s)
        }
    }

}
